src/openapi/transport/retry.js

```diff
@@ -130,29 +139,63 @@ TransportRetry.prototype.sendTransportCall = function(transportCall) {
  * @param transportCall
  */
 TransportRetry.prototype.addFailedCall = function(transportCall) {
-    transportCall.retryCount++;
     this.failedCalls.push(transportCall);
-    if (!this.retryTimer) {
+    const callOptions = this.methods[transportCall.method];
+    if (callOptions.retryTimeouts && callOptions.retryTimeouts.length > transportCall.retryCount) {
+        // schedule an individual retry timeout
+        transportCall.retryTimer = setTimeout(() => {
+            this.retryIndividualFailedCall(transportCall);
+        }, callOptions.retryTimeouts[transportCall.retryCount]);
+    } else if (!this.retryTimer) {
+        // schedule a retry timeout for all failed calls
         this.retryTimer = setTimeout(() => {

```

Comment:
I don’t think we should add it to the array of failed calls
because the only use it has is we have to filter it out when retryFailedCalls is called